packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -572,6 +570,49 @@ Future<XcodeBuildResult> buildXcodeProject({
   }
 }
 
+Future<RunResult> _runBuildWithRetries(List<String> buildCommands, BuildableIOSApp app) async {
+  int buildRetryDelaySeconds = 1;
+  int remainingTries = 5;
```

Comment:
I would give it 2 or 3 more tries, to cover the case where it needs to wait the entire duration of an existing xcode build.